src/borg/testsuite/compress.py

```diff
@@ -49,168 +52,122 @@ def test_lz4_buffer_allocation(monkeypatch):
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 50
-
-    cs = CompressionSpec("obfuscate,2,lz4")
-    assert isinstance(cs.inner.compressor, LZ4)
-    compressor = cs.compressor
-    data = bytes(10000)
-    _, compressed = compressor.compress({}, data)
-    min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
-    assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * 1001
-    # compressing 100 times the same data should give multiple different result sizes
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 10
-
-    cs = CompressionSpec("obfuscate,6,zstd,3")
-    assert isinstance(cs.inner.compressor, ZSTD)
-    compressor = cs.compressor
-    data = bytes(10000)
-    _, compressed = compressor.compress({}, data)
-    min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
-    assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * 10000001
-    # compressing 100 times the same data should give multiple different result sizes
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 90
-
-    cs = CompressionSpec("obfuscate,2,auto,zstd,10")
-    assert isinstance(cs.inner.compressor, Auto)
-    compressor = cs.compressor
-    data = bytes(10000)
-    _, compressed = compressor.compress({}, data)
-    min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
-    assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * 1001
-    # compressing 100 times the same data should give multiple different result sizes
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 10
-
-    cs = CompressionSpec("obfuscate,110,none")
-    assert isinstance(cs.inner.compressor, CNONE)
+@pytest.mark.parametrize(
+    "specs, c_type, result_range", [
+        ("obfuscate,1,none", CNONE, 50),
+        ("obfuscate,110,none", CNONE, None),
+        ("obfuscate,2,lz4", LZ4, 10),
+        ("obfuscate,6,zstd,3", ZSTD, 90),
+        ("obfuscate,2,auto,zstd,10", Auto, 10),
+    ]
+)
+def test_obfuscate(specs, c_type, result_range):
+    cs = CompressionSpec(specs)
+    assert isinstance(cs.inner.compressor, c_type)
     compressor = cs.compressor
-    data = bytes(1000)
-    _, compressed = compressor.compress({}, data)
-    assert 1000 <= len(compressed) <= 1000 + 1024
-    data = bytes(1100)
-    _, compressed = compressor.compress({}, data)
-    assert 1100 <= len(compressed) <= 1100 + 1024
+    data_bytes = bytes(10000)
+    _, compressed = compressor.compress({}, data_bytes)
+
+    # tests obfuscation with compression
+    if not isinstance(cs.inner.compressor, CNONE):
+        min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
+        assert max_compress * len(data_bytes) <= len(compressed) <= min_compress * len(data_bytes) * 1001
```

Comment:
that factor 1001 comes from obfuscation level 2, guess it should be parametrized also.

same for 101 (1) and 10000001 (6).